### DISCUSSION TITLE
echo listener with listenToAll

### DISCUSSION THREAD

**Discussion by patrickweh** [livewire/livewire#5030] [1. Help] [answered]
Hi,

i want to listen to all events on a channel without specifiying the event name.
With echo i could do something like this:

`Echo.private('App.Models.Calendar')
   .listenToAll((event, data) => {
      // do what you need to do based on the event name and data
      console.log(event, data)
   });`

is there a way to use listenToAll within the getListeners?

**Comment by joshhanley** [comment, marked as answer]
@patrickweh not at the moment. You would have to use the echo javascript client directly to set that up instead of using Livewire's listeners.

Hope this helps!
